pycue/opencue/cuelogging.py

```
@@ -0,0 +1,166 @@
+                    msg = e
+
+                if not os.access(log_dir, os.F_OK):
+                    err = "Unable to see log directory: %s, mkdir failed with: %s" % (
+                        log_dir, msg)
+                    raise RuntimeError(err)
+
+            if not os.access(log_dir, os.W_OK):
+                err = "Unable to write to log directory %s" % log_dir
+                raise RuntimeError(err)
+
+            try:
+                # Rotate any old logs to a max of MAX_LOG_FILES:
+                if os.path.isfile(self.filepath):
+                    rotateCount = 1
+                    while (os.path.isfile("%s.%s" % (self.filepath, rotateCount))
+                           and rotateCount < maxLogFiles):
+                        rotateCount += 1
+                    os.rename(self.filepath,
+                              "%s.%s" % (self.filepath, rotateCount))
+            # pylint: disable=broad-except
+            except Exception as e:
+                err = "Unable to rotate previous log file due to %s" % e
+                # Windows might fail while trying to rotate logs for checking if file is
+                # being used by another process. Frame execution doesn't need to
+                # be halted for this.
+                if platform.system() == "Windows":
+                    log.warning(err)
+                else:
+                    raise RuntimeError(err)
+            # pylint: disable=consider-using-with
+            self.fd = open(self.filepath, "w+", 1, encoding='utf-8')
+            try:
+                os.chmod(self.filepath, 0o666)
+            # pylint: disable=broad-except
+            except Exception as e:
+                err = "Failed to chmod log file! %s due to %s" % (self.filepath, e)
+                log.warning(err)
+        elif mode == MODE_READ:
+            self.fd = open(self.filepath, "r", encoding='utf-8')
+        else:
+            log.error("Unknown mode for log file!")
```

Simply logging an error at this point is not enough to prevent consumers of this class from using the class thinking it has been initialized properly. Please raise an exception.

---

Have split it into 2 separate classes, so should not be a concern anymore?